### PR TITLE
Add spring to rest

### PR DESCRIPTION
**Pull request by tetiana-ts**

### CODE REVIEW

File: rest/pom.xml [hunk 1]
```diff
@@ -34,6 +34,24 @@
       <groupId>ch.qos.logback</groupId>
       <artifactId>logback-classic</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
```
Comment (aleksashyn) [inline rest/pom.xml]: use jackson-databind instead core

File: rest/pom.xml [hunk 2]
```diff
@@ -34,6 +34,24 @@
...
+      <version>2.9.8</version>
+    </dependency>
+
+    <!--Spring-->
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context</artifactId>
```
Comment (aleksashyn) [inline rest/pom.xml]: spring-mvc already include this

File: rest/pom.xml [hunk 3]
```diff
@@ -34,6 +34,24 @@
       <groupId>ch.qos.logback</groupId>
       <artifactId>logback-classic</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
+      <version>2.9.8</version>
```
Comment (aleksashyn) [inline rest/pom.xml]: please specify all versions in root pom

File: rest/pom.xml [hunk 4]
```diff
@@ -56,6 +74,18 @@
...
     </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-test</artifactId>
+      <version>5.1.4.RELEASE</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
```
Comment (aleksashyn) [inline rest/pom.xml]: add scope provided

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.lms.system.teacher.crm.api;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@ComponentScan("com.lms.system.teacher.crm.api")
+@Configuration
+
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: please fix code formating

File: rest/src/main/java/com/lms/system/teacher/crm/api/Demo.java [hunk 6]
```diff
@@ -3,10 +3,12 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
 
 @Getter
 @AllArgsConstructor
 @NoArgsConstructor
+@Component ("DemoBean")
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/Demo.java]: bean name is redundant

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+public class RestConfiguration {
+
+    @Bean(name = "DemoBean")
+    public Demo getDemo() {
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: this bean already declare via Component

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.lms.system.teacher.crm.api;
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: move Configuration to package `config`

File: rest/src/main/webapp/WEB-INF/web.xml [hunk 9]
```diff
@@ -3,5 +3,4 @@
          xmlns="http://xmlns.jcp.org/xml/ns/javaee"
          xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
          version="3.1">
-
```
Comment (aleksashyn) [inline rest/src/main/webapp/WEB-INF/web.xml]: you need DispatcherServlet

File: rest/src/test/java/com/lms/system/teacher/crm/api/DemoTest.java [hunk 10]
```diff
@@ -4,9 +4,32 @@
...
 import org.junit.jupiter.api.Test;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
 class DemoTest {
 
+    @Test
+    void shouldReturnDemoBean() {
+
+        ApplicationContext context = new AnnotationConfigApplicationContext(RestConfiguration.class);
```
Comment (aleksashyn) [inline rest/src/test/java/com/lms/system/teacher/crm/api/DemoTest.java]: please use test config via annotations

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@ComponentScan("com.lms.system.teacher.crm.api")
+@Configuration
+
+public class RestConfiguration {
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: you need to declare Jackson bean

File: rest/pom.xml [hunk 12]
```diff
@@ -34,6 +34,24 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-webmvc</artifactId>
+      <version>5.1.4.RELEASE</version>
+    </dependency>
+
 
     <!--Test-->
     <dependency>
```
Comment (aleksashyn) [inline rest/pom.xml]: please add json-path for test purpose

File: rest/pom.xml (hunk 1 above)
Comment (tetiana-ts) [reply rest/pom.xml]: Ok

File: rest/pom.xml (hunk 2 above)
Comment (tetiana-ts) [reply rest/pom.xml]: Understood

File: rest/pom.xml (hunk 4 above)
Comment (tetiana-ts) [reply rest/pom.xml]: Ok

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java (hunk 5 above)
Comment (tetiana-ts) [reply rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: Ok

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java (hunk 8 above)
Comment (tetiana-ts) [reply rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: Ok!

File: rest/src/main/webapp/WEB-INF/web.xml (hunk 9 above)
Comment (tetiana-ts) [reply rest/src/main/webapp/WEB-INF/web.xml]: created

File: rest/src/test/java/com/lms/system/teacher/crm/api/DemoTest.java (hunk 10 above)
Comment (tetiana-ts) [reply rest/src/test/java/com/lms/system/teacher/crm/api/DemoTest.java]: Ok

File: rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java (hunk 11 above)
Comment (tetiana-ts) [reply rest/src/main/java/com/lms/system/teacher/crm/api/RestConfiguration.java]: Ok

File: rest/pom.xml (hunk 12 above)
Comment (tetiana-ts) [reply rest/pom.xml]: Ok

File: rest/pom.xml (hunk 3 above)
Comment (tetiana-ts) [reply rest/pom.xml]: Ok!

File: pom.xml [hunk 13]
```diff
@@ -3,253 +3,254 @@
          xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
-  <modelVersion>4.0.0</modelVersion>
```
Comment (aleksashyn) [inline pom.xml]: please change you indentation for xml file to avoid a lot of changes

File: rest/pom.xml [hunk 14]
```diff
@@ -2,85 +2,130 @@
...
+            <artifactId>lombok</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>[2.0.0-RC1,2.9.8]</version>
```
Comment (aleksashyn) [inline rest/pom.xml]: try to use only stable versions

File: rest/pom.xml [hunk 15]
```diff
@@ -2,85 +2,130 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
```
Comment (aleksashyn) [inline rest/pom.xml]: please extract version to root pom

File: rest/pom.xml [hunk 16]
```diff
@@ -2,85 +2,130 @@
...
-        <groupId>org.jacoco</groupId>
-        <artifactId>jacoco-maven-plugin</artifactId>
-      </plugin>
-    </plugins>
-  </build>
+        <!--Spring-->
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-webmvc</artifactId>
+            <version>5.1.4.RELEASE</version>
```
Comment (aleksashyn) [inline rest/pom.xml]: the same

File: rest/pom.xml [hunk 17]
```diff
@@ -2,85 +2,130 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <version>5.1.4.RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
+            <artifactId>json-path</artifactId>
+            <version>2.4.0</version>
```
Comment (aleksashyn) [inline rest/pom.xml]: please add scope test

File: rest/spotbugs-exclude.xml [hunk 18]
```diff
@@ -0,0 +1,461 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aleksashyn) [inline rest/spotbugs-exclude.xml]: you need to exclude ONLY you case

File: rest/src/main/webapp/WEB-INF/spring/dispatcher-servlet.xml [hunk 19]
```diff
@@ -0,0 +1,13 @@
+<beans xmlns="http://www.springframework.org/schema/beans"
```
Comment (aleksashyn) [inline rest/src/main/webapp/WEB-INF/spring/dispatcher-servlet.xml]: please do not use xml configuration

File: rest/src/main/java/com/lms/system/teacher/crm/api/config/Initializer.java [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.web.servlet.DispatcherServlet;
+
+public class Initializer implements WebApplicationInitializer {
+
+
+    private static final String DISPATCHER_SERVLET_NAME = "dispatcher";
+
+    @Override
+    public void onStartup(ServletContext servletContext) {
+        AnnotationConfigWebApplicationContext ctx = new AnnotationConfigWebApplicationContext();
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/config/Initializer.java]: Please keep servlet into web.xml and just add init param that can resolve java spring config

File: rest/src/main/java/com/lms/system/teacher/crm/api/config/RestConfiguration.java [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+public class RestConfiguration {
+
+    @Bean
+    public Demo demo() {
+        return new Demo();
+    }
+
+    @Bean
+    public ObjectMapper mapper() {
+        return new ObjectMapper();
```
Comment (aleksashyn) [inline rest/src/main/java/com/lms/system/teacher/crm/api/config/RestConfiguration.java]: Please, connect jekson with spring

File: mvnw.cmd [hunk 22]
```diff
@@ -1,161 +1,161 @@
-@REM ----------------------------------------------------------------------------
```
Comment (aleksashyn) [inline mvnw.cmd]: may be you build failed after this changes

**Pull request merged**